### PR TITLE
Add nyc to simplify test/coverage commands

### PR DESCRIPTION
**Pull request by JaKXz**
https://circle-artifacts.com/gh/rangle/typescript-react-redux-starter/496/artifacts/0/home/ubuntu/typescript-react-redux-starter/coverage/lcov-report/index.html

- Renamed `test-utils` to `test` so it’s picked up automatically by
mocha
- Using `_mocha` executable which exposes coverage/instrumentation
things with `ts-node` as before
- better lint task
- better dev task (`-k` flag kills other processes when one dies)
- better test:watch task
- add `--bail` to build task to fail fast
- add arbitrary coverage thresholds @ 80%
- add coverage report to CircleCI artifacts

Fixes rangle/rangle-starter#112. 
Connected to rangle/rangle-starter#12.
Connected to rangle/rangle-starter#8. 
Distant cousin of rangle/react-redux-starter#184.

Can @SethDavenport review?

### CODE REVIEW

**Comment by JaKXz** [comment]
@clbond I'm guessing your work can apply to this project as well? Feel free to close this PR if so :)

**Comment by clbond** [comment]
@JaKXz I did fix the source maps and change the target to ES6 / add Babel in this PR - https://github.com/rangle/typescript-react-redux-starter/pull/146 - but it sounds like your work is orthogonal to that. I did not make any changes to the coverage report in typescript-react-redux-starter. In fact I could not figure out how to even generate a coverage report if I am being honest! It was a lot more straightforward in the Angular2 starters. But I think we can merge both of our PRs without issue.

File: circle.yml [hunk 1]
```diff
@@ -15,11 +15,12 @@ dependencies:
 
 test:
   post:
-    - npm run build
+    - npm run build # NOTE: this is run twice in total because of the postinstall step
```
Comment (SethDavenport) [inline circle.yml]: Yeah we probably don't need it here then.

File: package.json [hunk 2]
```diff
@@ -8,21 +8,35 @@
...
     "prebuild": "npm run clean",
-    "build": "cross-env NODE_ENV=production webpack -p",
+    "build": "cross-env NODE_ENV=production webpack -p --bail",
     "clean": "rimraf dist/",
     "dev:server": "cross-env NODE_ENV=development PORT=3000 nodemon server/node-server.js",
     "dev:client": "cross-env NODE_ENV=development webpack-dev-server -d --inline --hot --progress --no-info",
-    "dev": "concurrently -r 'npm run dev:client' 'npm run dev:server'",
-    "lint": "npm run lint-ts && npm run lint-js && npm run lint-css",
+    "dev": "concurrently -kr 'npm run dev:client' 'npm run dev:server'",
+    "lint": "concurrently 'npm run lint-ts' 'npm run lint-js' 'npm run lint-css'",
```
Comment (SethDavenport) [inline package.json]: You know what I'm going to say :)

File: tsconfig.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "experimentalDecorators": true,
     "noImplicitAny": false,
     "removeComments": false,
-    "sourceMap": true,
+    "inlineSourceMap": true,
```
Comment (SethDavenport) [inline tsconfig.json]: Need to make sure this doesn't explode bundle size from `npm run build`...

**Comment by JaKXz** [comment]
@clbond @SethDavenport this is not perfect, so I'd suggest trying the approach from rangle/angular2-redux-starter#180 instead. Closing for now, though I'm not sure rangle/rangle-starter#12 should be closed just yet [until some solution is found for these things].

**Pull request closed without merging**